### ISSUE TITLE
[12.1.0+] Comments with fixed columns throw error and show comment box under the table when table is scrolled horizontally

### ISSUE THREAD

**Issue by AMBudnik** [handsontable/handsontable#9645]
### Description

Possibly related to changes in https://github.com/handsontable/handsontable/issues/4785. I've connected the issue to pinpoint the difference in the result. But it does not seem to work well in any of those versions.

Before 12.1.0 the comment box was misplaced while scrolling horizontally and eventually while scrolling right it entirely disappeared from the view. After that action, we got the following console error

```js
Uncaught TypeError: Cannot read properties of undefined (reading 'defaultView')
    at offset (handsontable.full.js:2881:33)
    at Comments.refreshEditor (handsontable.full.js:142750:44)
    at Comments.show (handsontable.full.js:142665:12)
    at Comments.showAtCell (handsontable.full.js:142688:19)
    at DisplaySwitch.<anonymous> (handsontable.full.js:142430:23)
    at fastCall (handsontable.full.js:8464:17)
    at DisplaySwitch.runLocalHooks (handsontable.full.js:14030:32)
    at DisplaySwitch.<anonymous> (handsontable.full.js:143524:18)
    at handsontable.full.js:8277:21
```

In 12.1.0 while we scroll right we can see the same misplacement but eventually, the comment box jumps after the table DIV (appears under the table) and we get a different error in the console.

```js
handsontable.full.js:148815 Uncaught TypeError: TD.getBoundingClientRect is not a function
    at Comments.refreshEditor (handsontable.full.js:148815:38)
    at Comments.show (handsontable.full.js:148728:12)
    at Comments.showAtCell (handsontable.full.js:148745:19)
    at DisplaySwitch.<anonymous> (handsontable.full.js:148488:23)
    at fastCall (handsontable.full.js:8008:17)
    at DisplaySwitch.runLocalHooks (handsontable.full.js:12936:32)
    at DisplaySwitch.<anonymous> (handsontable.full.js:149613:18)
    at handsontable.full.js:7821:21
```

### Steps to reproduce

1. Scroll the table right to hide `T` column
2. Hoover over `B2` cell to show the comment

#### result for pre 11.1.0 and older
![Screenshot 2022-07-11 at 14 13 42](https://user-images.githubusercontent.com/10757813/178261497-1b09cfd4-981d-42df-842c-a714c4068be1.png)

![Screenshot 2022-07-11 at 14 12 11](https://user-images.githubusercontent.com/10757813/178261214-f42cebec-2ef3-4ac2-9af4-38f29baa6bc9.png)

#### result for 12.1.0+
![Screenshot 2022-07-11 at 13 10 04](https://user-images.githubusercontent.com/10757813/178260666-37e1847c-a865-4e84-b301-c306302f2a3b.png)
![Screenshot 2022-07-11 at 13 10 20](https://user-images.githubusercontent.com/10757813/178260671-ae5eef41-3a2e-4e9f-b5aa-22c167153320.png)

### Demo

https://jsfiddle.net/vmkLjbod/ - 12.1.0
https://jsfiddle.net/vmkLjbod/1/ - 11.1.0

### Your environment
* Handsontable version: 12.1.0+
* Browser name and version: Chrome 103
* Operating system: mcOS Monretey

Inform [Zen](https://handsontable.zendesk.com/agent/tickets/29591), [Zen](https://handsontable.zendesk.com/agent/tickets/29957)

**Comment by wszymanski** [comment]
After some investigation I've found that we have 4 `getCell` methods (`handsontable/src/3rdparty/walkontable/src/facade/core.js`, `handsontable/src/3rdparty/walkontable/src/table.js`,  `handsontable/src/3rdparty/walkontable/src/core/_base.js` and `handsontable/src/core.js`) and for unknown reason in the below code, there is used `getCell` method on `hot.view._wt.wtTable` instead of calling `this.hot.getCell(renderableRow, renderableColumn, true)`.

https://github.com/handsontable/handsontable/blob/3397ce5ca20a640877710e3f1e935b4106754ee7/handsontable/src/plugins/comments/comments.js#L472-L475

It's worth to mention that already used method is taking cells from the `MasterTable` which after scrolling doesn't contain cells already placed on left overlay (after using `fixedColumnsLeft` option). Another `getCell` methods provide as the last, a `topMost` argument which controls whether cell should be taken from proper (sometimes another) overlay. 

Taking cell from the `MasterTable` is reason of the bug (we get an error code, instead of DOM element).

https://github.com/handsontable/handsontable/blob/a2033859014e4e115cf54e638b23727f91c07f29/handsontable/src/3rdparty/walkontable/src/table.js#L616-L618

Calling the `getBoundingClientRect` method on number (instead of expected, by code's author, DOM element) throws an error.

https://github.com/handsontable/handsontable/blob/3397ce5ca20a640877710e3f1e935b4106754ee7/handsontable/src/plugins/comments/comments.js#L489-L494

The above suggestion could be a solution. However, #6953 introduced two tests which allow users to show comments on already hidden cell. After discussion with @jansiegel we figured it could be changed if hasn't came from some business need. I need a little bit more time for discussing it and preparing code change.
